OneSchedule.Tests/OnesheduleStateTests.cs

```diff
@@ -0,0 +1,225 @@
+using Moq;
+using NUnit.Framework;
+using OneSchedule.Data.Abstractions;
+using OneSchedule.Domain.Abstractions.StateMachine;
+using OneSchedule.Domain.Models;
+using OneSchedule.Domain.StateMachine.AskState;
+using OneSchedule.Domain.StateMachine.GetState;
+using OneSchedule.Entities;
+using OneSchedule.Exceptions.CustomExceptions;
+using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+using System.Threading.Tasks;
+using Telegram.Bot;
+using Telegram.Bot.Types.Enums;
+using Telegram.Bot.Types.ReplyMarkups;
+
+namespace OneSchedule.Tests
+{
+    public class OnesheduleStateTests
+    {
+        private Mock<ITelegramBotClient> _botMoq;
+        private readonly Mock<IStateContext> _stateContextMoq;
+
+        public OnesheduleStateTests()
+        {
+            _botMoq = new Mock<ITelegramBotClient>();
+            _stateContextMoq = new Mock<IStateContext>();
+        }
+
+        [Test]
+        public async Task AskBeginDateState_IsCompleted()
+        {
+            _botMoq = new Mock<ITelegramBotClient>();
+            var chatId = "123";
+            var state = new AskBeginDateState(_botMoq.Object);
```

Comment:
It is better to move this part to a constructor, because you use the same object for other related tests or move closer to Act part inside the test to make the purpose of this line more clear